### PR TITLE
Init static site rendering

### PR DESCRIPTION
**Pull request by sebastianruehmann**
Added static site rendering for visitors on area specific links

### CODE REVIEW

**Comment by jonicious** [comment]
I really like the idea! When playing around with it locally, I noticed a few issues.

For Hamburg, I got the following error in the dev console:
>Warning: Text content did not match. Server: "74.81" Client: "74,81"

Maybe because we use `toLocaleString`.

http://localhost:3000/s/Berlin%20Mitte gives me a 404. http://localhost:3000/s/Hamburg however works.

At the moment, https://hotspotornot.de/s/Berlin redirects to https://hotspotornot.de/s/Berlin%20Mitte. This is no longer the case. So now only areas available in the RKI API work. I kind of like the flexibility right now but it does not matter too much.

File: src/pages/s/[area].js [hunk 1]
```diff
@@ -1,75 +1,71 @@
 import React, { useEffect } from 'react'
-import { useRouter } from 'next/router'
 import { useQuery } from 'react-query'
-import { getCoordinatesByAddress } from '../../services/Geocoding'
-
+import styled from 'styled-components'
+import axios from 'axios'
```
Comment (jonicious) [inline src/pages/s/[area].js:4]: FYI: There are some unused imports in the file, this is what it looks like in IntelliJ.

IntelliJ

File: src/pages/s/[area].js [hunk 2]
```diff
@@ -1,75 +1,71 @@
...
 
-        <Footer />
-      </>
-    )
-  }
+export async function getStaticProps(props) {
+  const result = await searchByArea(props.params.area)
 
-  return null
+  return { props: { result }, revalidate: 30 }
```
Comment (jonicious) [inline src/pages/s/[area].js:68]: The 30 could also be 1, right?

File: src/services/Api.js [hunk 3]
```diff
@@ -2,41 +2,86 @@ import axios from 'axios'
...
+        (feature) => feature.attributes.GEN
+      )
+      return areas
+    })
+}
+
+export function searchByArea(area) {
+  return axios
+    .get(
+      `${process.env.NEXT_PUBLIC_RKI_API}?where=GEN = '${encodeURIComponent(
```
Comment (jonicious) [inline src/services/Api.js:71]: Is the space in the URL necessary?

File: src/services/Api.js [hunk 4]
```diff
@@ -2,41 +2,86 @@ import axios from 'axios'
...
+        where: '1=1',
+      },
+    })
+    .then((response) => {
+      if (!response.data.features.length) {
+        throw new Error('No area data found')
       }
+
+      const areas = response.data.features.map(
+        (feature) => feature.attributes.GEN
```
Comment (jonicious) [inline src/services/Api.js:62]: Could `attributes` potentially be `null`?

File: src/services/Api.js [hunk 5]
```diff
@@ -2,41 +2,86 @@ import axios from 'axios'
...
-  geometryType: 'esriGeometryPoint',
-  spatialRel: 'esriSpatialRelWithin',
-  inSR: '4326',
-  outSR: '4326',
+  returnGeometry: false,
   f: 'json',
 }
 
-//  TODO: Rename fields in here.
+const formatResponse = (response) => {
```
Comment (jonicious) [inline src/services/Api.js:9]: Sometimes, we use good old function, sometimes arrow functions. Maybe we should be consistent? I don't really care though  ¯\_(ツ)_/¯

File: src/services/Api.js [hunk 6]
```diff
@@ -2,41 +2,86 @@ import axios from 'axios'
...
+        outFields: 'GEN',
+        where: '1=1',
+      },
+    })
+    .then((response) => {
+      if (!response.data.features.length) {
+        throw new Error('No area data found')
       }
+
+      const areas = response.data.features.map(
```
Comment (jonicious) [inline src/services/Api.js:61]: Could directly return the `areas`

File: src/pages/s/[area].js (hunk 2 above)
Comment (sebastianruehmann) [reply src/pages/s/[area].js:68]: A stale duration from 30 sec is fine for our data IMO

File: src/services/Api.js (hunk 3 above)
Comment (sebastianruehmann) [reply src/services/Api.js:71]: That's the formatting aligned to the official docs 🤷

File: src/services/Api.js (hunk 4 above)
Comment (sebastianruehmann) [reply src/services/Api.js:62]: Ney 👍

File: src/services/Api.js (hunk 5 above)
Comment (sebastianruehmann) [reply src/services/Api.js:9]: Let's make good old functions to our policy? :)

File: src/pages/s/[area].js (hunk 1 above)
Comment (sebastianruehmann) [reply src/pages/s/[area].js:4]: 👍

**Pull request closed without merging**